src/Compilers/CSharp/Test/Symbol/Compilation/SemanticModelGetSemanticInfoTests.cs

```diff
@@ -13062,12 +13062,11 @@ public static void Main()
             var semanticInfo = GetSemanticInfoForTest<IdentifierNameSyntax>(sourceCode);
 
             Assert.Null(semanticInfo.Type);
-            Assert.Equal("?", semanticInfo.ConvertedType.ToTestDisplayString());
-            Assert.Equal(TypeKind.Error, semanticInfo.ConvertedType.TypeKind);
+            Assert.Null(semanticInfo.ConvertedType);
             Assert.Equal(ConversionKind.Identity, semanticInfo.ImplicitConversion.Kind);
 
             Assert.Null(semanticInfo.Symbol);
-            Assert.Equal(CandidateReason.NotInvocable, semanticInfo.CandidateReason);
+            Assert.Equal(CandidateReason.OverloadResolutionFailure, semanticInfo.CandidateReason);
```

Comment:
Is this really an overload resolution failure? This change feels unexpected, what caused it?

Reply:
> Is this really an overload resolution failure? This change feels unexpected, what caused it?

This is an error scenario, we cannot bind a method group in this context. The change is cased by appropriately marking a bound node as compiler generated (I think its the change in BindCollectionInitializerElement). The change in SemanticModel behavior feels acceptable to me.

---
In reply to: [147787383](https://github.com/dotnet/roslyn/pull/22894#discussion_r147787383) [](ancestors = 147787383)